### PR TITLE
Alerting: Add RBAC actions and role for provisioning API routes

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:
Currently, in order to access alerting via provisioning API user must have admin privileges (role Admin). These are too relaxed permissions as they basically let the user (or API key) very broad access to any Grafana API.
The reason for that is that the provisioning API allows users to access alert rules that belong to a protected folder.

This PR introduces RBAC actions and a role that will allow users (for now just enterprise ones) to restrict access to just provisioning API. 
Actions: 
- `alert.provisioning:read`
- `alert.provisioning:write`
Both actions do not require scope and applied to the entire organization.

Role: `fixed:alerting:provisioning` that lets user that has that role access only routes that belong to the provisioning API. This role is assigned to the OSS Admin role.

### CODE REVIEW

**Review by santihernandezc — APPROVED** [review]
LGTM, I would wait for feedback in the docs part before merging.

**Review by alexweav — APPROVED** [review]
LGTM, thanks @yuri-tceretian !

**Comment by alexweav** [comment]
Build problem is a minor docs linting thing

**Pull request merged**
